ross/rotor_assembly.py

```diff
@@ -512,6 +518,35 @@ def flatten(l):
 
         self.df = df
 
+    def _check_number_dof(self):
+        number_dof = len(self.shaft_elements[0].dof_mapping()) / 2
+
+        for el in self.shaft_elements:
+            if (len(el.dof_mapping()) == 8 and number_dof == 6) or (
+                len(el.dof_mapping()) == 12 and number_dof == 4
+            ):
+                raise Exception(
+                    "The number of degrees o freedom of all elements must be the same! There are SHAFT elements with discrepant DoFs."
+                )
+
+        for el in self.disk_elements:
+            if (len(el.dof_mapping()) == 4 and number_dof == 6) or (
+                len(el.dof_mapping()) == 6 and number_dof == 4
+            ):
+                raise Exception(
+                    "The number of degrees o freedom of all elements must be the same! There are DISK elements with discrepant DoFs."
+                )
+
+        for el in self.bearing_elements:
+            if (len(el.dof_mapping()) == 2 and number_dof == 6) or (
+                len(el.dof_mapping()) == 3 and number_dof == 4
+            ):
+                raise Exception(
+                    "The number of degrees o freedom of all elements must be the same! There are BEARING elements with discrepant DoFs."
+                )
+
+        return int(number_dof)
```

Comment:
We could do it shorter using python's `any()` method.
Please, correct me if there is a problem with the logic.

```
    def _check_number_dof(self):
        """Verify the consistency of degrees of freedom.

        This method loops for all the elements, checking if the number of degrees of
        freedom is consistent.
        E.g.: inputting 2 shaft elements, one with 4 dof and one with 6, will raise
        an error.

        Raises
        ------
        Exception
            Error pointing out difference between the number of DoF's from each element
            type.

        Returns
        -------
        number_dof : int
            Number of degrees of freedom from the adopted shaft element.
        """
        number_dof = len(self.shaft_elements[0].dof_mapping())

        if any(len(sh.dof_mapping()) != number_dof for sh in self.shaft_elements):
            raise Exception(
                "The number of degrees o freedom of all elements must be the same! There are SHAFT elements with discrepant DoFs."
            )

        if any(len(disk.dof_mapping()) != number_dof / 2 for disk in self.disk_elements):
            raise Exception(
                "The number of degrees o freedom of all elements must be the same! There are DISK elements with discrepant DoFs."
            )

        if any(len(brg.dof_mapping()) != number_dof / 4 for brg in self.bearing_elements):
            raise Exception(
                "The number of degrees o freedom of all elements must be the same! There are BEARING elements with discrepant DoFs."
            )

        return int(number_dof / 2)
```

Reply:
Nice! I did not know about the "any()" method. There is a small correction to be made though, the number_dof would be *2, itself and /2, instead of itself, /2 and /4. Other than that i think it would work. We will be analyzing this code and implementing it, thanks again!

Reply:
```
number_dof = len(self.shaft_elements[0].dof_mapping())
```
number_dof would return 8 or 12 in this code line. The `dof_mapping()` for shaft elements returns de DoF's for 2 nodes. So, its value ends up being doubled.

```len(disk.dof_mapping())``` returns 4 or 6 and ```len(brg.dof_mapping())``` returns 2 or 3. We compare the number_dof, dividing itself by 2 or 4, or multiplying the `len` by 2 or 4.

Reply:
Oh, I missed what you've done before. On your code, you already divided the `number_dof` by 2 in `number_dof = len(self.shaft_elements[0].dof_mapping()) / 2`. Sorry.